### PR TITLE
Fix reconciler issue while creating knative service

### PR DESCRIPTION
**Pull request by savitaashture**
# Changes

Fix: https://github.com/tektoncd/triggers/issues/983

**Root Cause:**
The return type of `unstructured.NestedFieldCopy` is `map[string]interface{}` in code we were comparing with `map[string]string` and because of that annotations were not matching.
Knative adds strict check on default annotation so because of that validation was failing.

/cc @dibyom @Fabian-K 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#tests) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#docs) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commits)
- [ ] Release notes block has been filled in or deleted (only if no user facing changes)

_See [the contribution guide](https://github.com/tektoncd/triggers/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by savitaashture** [comment]
it seems some infra issues

File: pkg/dynamic/custom_reconcile.go [hunk 1]
```diff
@@ -238,6 +238,19 @@ func MergeMaps(m1, m2 map[string]string) map[string]string {
 	return merged
 }
 
+// mergeMaps merges the values in the passed maps into a new map.
+// Values within m2 potentially clobber m1 values.
+func mergeMaps(m1, m2 map[string]interface{}) map[string]interface{} {
```
Comment (dibyom) [inline pkg/dynamic/custom_reconcile.go:243]: small nit: we already have a function called MergeMaps, so maybe call this mergeStringInterfaceMaps (or call the other one mergeStringMaps)?

**Comment by dibyom** [comment]
Also: I don't know how hard it is but could we add an unit test to verify the fix?

**Comment by savitaashture** [comment]
@dibyom i have refactored and updated code
So now we don't require extra `MergeMaps` function

**Comment by dibyom** [comment]
We don't have to do this for this PR but we should track adding a full e2e test for the knative EL (perhaps using https://github.com/csantanapr/knative-kind and the e2e kind stuff that Billy set up)

**Comment by savitaashture** [comment]
> We don't have to do this for this PR but we should track adding a full e2e test for the knative EL (perhaps using https://github.com/csantanapr/knative-kind and the e2e kind stuff that Billy set up)

+1 
right now if i add e2e the pre req would be cluster should have Knative running 

But as you said we should do this sometime

**Comment by dibyom** [comment]
> right now if i add e2e the pre req would be cluster should have Knative running

Let's open an issue to track this (Billy has been doing some work on running e2e tests using kind -- we could piggy back off of that)

**Pull request merged**